### PR TITLE
replace tranpose with reshape

### PR DESCRIPTION
**Pull request by duli2012**
If Transpose is NCHW->NHWC or NHWC->NCHW, replace transpose with reshape in one of the two cases:
1)C==1 
2) H==W==1

### CODE REVIEW

**Comment by pengwa** [comment]
I am curious does this help in better perf?

**Comment by guschmue** [comment]
reshape in most implementations doesn't need to copy ... should help.

**Comment by guschmue** [comment]
@duli2012, can you make pylint happy:
https://travis-ci.org/onnx/tensorflow-onnx/builds/452726892?utm_source=github_status&utm_medium=notification

File: tf2onnx/optimizer/transpose_optimizer.py [hunk 1]
```diff
@@ -79,6 +79,31 @@ def pre_optimize_action(self):
...
+        # if channel==1 or height==width==1, replace transpose with reshape
+        for op in nodes:
+            if op.type == "Transpose":
+                input_shape = self._g.get_shape(op.input[0])
+                new_shape = []
+                if is_nchw_transpose(op) and (input_shape[3] == 1 or (input_shape[1] == 1 and input_shape[2] == 1)):
+                    new_shape = [input_shape[0], input_shape[3], input_shape[1], input_shape[2]]
+                if is_nhwc_transpose(op) and (input_shape[1] == 1 or (input_shape[2] == 1 and input_shape[3] == 1)):
+                    new_shape = [input_shape[0], input_shape[2], input_shape[3], input_shape[1]]
+                if len(new_shape) == 4:
```
Comment (guschmue) [inline tf2onnx/optimizer/transpose_optimizer.py]: you can just use ```if new_shape:```

File: tf2onnx/optimizer/transpose_optimizer.py [hunk 2]
```diff
@@ -79,6 +79,31 @@ def pre_optimize_action(self):
...
 
     def post_optimize_action(self):
+        nodes = self.nodes
+        # if channel==1 or height==width==1, replace transpose with reshape
+        for op in nodes:
+            if op.type == "Transpose":
+                input_shape = self._g.get_shape(op.input[0])
+                new_shape = []
+                if is_nchw_transpose(op) and (input_shape[3] == 1 or (input_shape[1] == 1 and input_shape[2] == 1)):
+                    new_shape = [input_shape[0], input_shape[3], input_shape[1], input_shape[2]]
```
Comment (guschmue) [inline tf2onnx/optimizer/transpose_optimizer.py]: ```if is_nchw_transpose``` and than transpose to 0,3,1,2 (aka NHWC_TO_NCHW) ... this is correct? 
A little comment would help.

File: tf2onnx/optimizer/transpose_optimizer.py [hunk 3]
```diff
@@ -79,6 +79,31 @@ def pre_optimize_action(self):
...
+                if is_nhwc_transpose(op) and (input_shape[1] == 1 or (input_shape[2] == 1 and input_shape[3] == 1)):
+                    new_shape = [input_shape[0], input_shape[2], input_shape[3], input_shape[1]]
```
Comment (guschmue) [inline tf2onnx/optimizer/transpose_optimizer.py]: Same as above.

File: tf2onnx/optimizer/transpose_optimizer.py (hunk 1 above)
Comment (duli2012) [reply tf2onnx/optimizer/transpose_optimizer.py]: Done.

File: tf2onnx/optimizer/transpose_optimizer.py (hunk 2 above)
Comment (duli2012) [reply tf2onnx/optimizer/transpose_optimizer.py]: Added

File: tf2onnx/optimizer/transpose_optimizer.py (hunk 3 above)
Comment (duli2012) [reply tf2onnx/optimizer/transpose_optimizer.py]: Added

**Pull request merged**